### PR TITLE
[`JDBC`] Change default `Precison` and `DisplaySize` for all datetime types.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury-Fridlyand <yuryf@bitquilltech.com>

### Description
The proposed fix changes default [`Precision`](https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getPrecision-int-) and [`DisplaySize`](https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getColumnDisplaySize-int-) configured for `TIME` and `DATE` types.

https://github.com/Bit-Quill/opensearch-project-sql/blob/522901d0ce18ff36c54eb1acf10e3d2a0b9381e4/sql-jdbc/src/main/java/org/opensearch/jdbc/types/OpenSearchType.java#L146-L147
`Precision` is the third arg and `DisplaySize` is the fourth arg.

-----
https://github.com/Bit-Quill/opensearch-project-sql/blob/522901d0ce18ff36c54eb1acf10e3d2a0b9381e4/sql-jdbc/src/main/java/org/opensearch/jdbc/types/OpenSearchType.java#L65-L66
Please, note, classes referenced here `^` are `java.sql.*` classes, not `java.time.*` ones. `Time` has no fraction seconds part.

### Test
**Before**
image

**After**
image

### Issues Resolved
opensearch-project/sql-jdbc#21
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by margarit-h — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

**Review by forestmvey — APPROVED** [review]

**Comment by GabeFernandez310** [comment]
I just had a general question. What unit of measurement are `precision` and `size` (i.e., is it the number of bits, bytes, digits, etc... )?

**Review by GabeFernandez310 — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
> I just had a general question. What unit of measurement are `precision` and `size` (i.e., is it the number of bits, bytes, digits, etc... )?

According to [docs](https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getPrecision-int-):
> For datetime datatypes, this is the length in characters of the String representation (assuming the maximum allowed precision of the fractional seconds component)

File: sql-jdbc/src/main/java/org/opensearch/jdbc/types/OpenSearchType.java [hunk 1]
```diff
@@ -62,8 +62,8 @@ public enum OpenSearchType {
     IP(JDBCType.VARCHAR, String.class, 15, 0, false),
     NESTED(JDBCType.STRUCT, null, 0, 0, false),
     OBJECT(JDBCType.STRUCT, null, 0, 0, false),
-    DATE(JDBCType.DATE, Date.class, 24, 24, false),
-    TIME(JDBCType.TIME, Time.class, 24, 24, false),
+    DATE(JDBCType.DATE, Date.class, 10, 10, false),
+    TIME(JDBCType.TIME, Time.class, 8, 8, false),
     DATETIME(JDBCType.TIMESTAMP, Timestamp.class, 24, 24, false),
     TIMESTAMP(JDBCType.TIMESTAMP, Timestamp.class, 24, 24, false),
```
Comment (acarbonetto) [inline sql-jdbc/src/main/java/org/opensearch/jdbc/types/OpenSearchType.java:68]: Should we update DATETIME and TIMESTAMP to a precision of `9`?

Comment (Yury-Fridlyand) [reply sql-jdbc/src/main/java/org/opensearch/jdbc/types/OpenSearchType.java]: Done, thanks!

**Review by MitchellGale-BitQuill — APPROVED** [review]

**Review by MaxKsyunz — APPROVED** [review]

**Pull request merged**